R/utilities.R

```diff
@@ -1148,3 +1148,185 @@ ToNumeric <- function(x){
   }
   return(x)
 }
+
+#' Export Seurat object for UCSC cell browser
+#'
+#' @param object Seurat object
+#' @param dir output directory path
+#' @param dataset.name name of the dataset
+#' @param meta.fields vector of metadata fields to export
+#' @param meta.fields.names list that defines metadata field names
+#'                          after the export. Should map metadata
+#'                          column name to export name
+#' @param embeddings vector of embedding names to export
+#' @param markers.file path to file with marker genes
+#' @param cluster.field name of the metadata field containing cell cluster
+#' @param cb.dir in which dir to create UCSC cell browser content root
+#' @param port \emph{experimental} on which port to run UCSC cell browser after export
+#'
+#' @export
+#'
+#' @examples
+#' ExportToCellbrowser(object = pbmc_small, dataset.name = "PBMC", dir = "out")
+#'
+ExportToCellbrowser <- function(
+  object,
+  dir = NULL,
+  dataset.name = NULL,
+  meta.fields = NULL,
+  meta.fields.names = NULL,
+  embeddings = c("tsne"),
+  markers.file = NULL,
+  cluster.field = NULL,
+  port = NULL,
+  cb.dir = NULL
+) {
+  if (is.null(dir)) {
+    stop("Missing output directory")
+  }
+  if (is.null(dataset.name)) {
+    stop("Missing dataset name parameter")
+  }
+  if (is.null(meta.fields)) {
+    meta.fields <- c("nCount_RNA", "nFeature_RNA")
+    if (length(levels(x = Idents(object = object))) > 1) {
+      meta.fields <- c(meta.fields, ".ident")
+    }
+  }
+  if (!is.null(port) && is.null(cb.dir)) {
+    stop("cb.dir parameter is needed when port is set")
+  }
+  if (!dir.exists(dir)) {
+    dir.create(dir)
+  }
+  if (!dir.exists(dir)) {
+    stop(sprintf("Output director %s cannot be created or is a file", dir))
```

Comment:
Functions like `stop` will automatically concatenate strings. We would recommend writing this as `stop("Output directory ", dir, " cannot be created or is a file")`